### DISCUSSION TITLE
How do we install Language satellite assmeblie

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#1250] [Q&A]
I have modified my solution to produce a satellite language for my module the package is created but when extracted it is in not being installed to the correct culture folder but to the root with the module files. Currently in my case its "nl-NL", manually moving the file to the folder and everything is ok.

My .nl-NL.1.0.0.nuspec file is just a copy of module file with one tag added    <language>nl-NL</language>.

**Comment by leigh-pointer** [comment]
Should add, I based everything on the MS docs [https://docs.microsoft.com/en-us/nuget/create-packages/creating-localized-packages](https://docs.microsoft.com/en-us/nuget/create-packages/creating-localized-packages)

**Comment by sbwalker** [comment]
Oqtane does not support the installation of satellite assemblies yet. I mentioned that in this thread https://github.com/oqtane/oqtane.translations/issues/16

**Comment by sbwalker** [comment]
InstallationManager - InstallPackages() needs to be enhanced. I would assume the satellite assemblies would be included within the nuget package in their natural folder structure:

lib/assemblyname.dll
lib/nl-NL/assemblyname.resources.dll
lib/fr-FR/assemblyname.resources.dll

does this sound correct?

**Reply by leigh-pointer** [reply]
Installed the satellites to lib/nl-NL/assemblyname.resources.dll
that went well
Moving the module out of the root to
lib/assemblyname.dll
no good.

**Reply by hishamco** [reply]
The problem with module is the developer need to modify the scripts in .cmd files to move the satellite assemblies to the `Oqtane.Server` binaries. I did few discussion with @sbwalker previously NuGet packages may solve this issue in professional way

I may hack the .cmd files to make this work if I got a time

**Reply by leigh-pointer** [reply]
@hishamco Yes I modified the cmd files and the nuspec file.  The files are built and the also the module resource. all files get added to the package.  I currently running my compiled site on IIS locally.  IIS has no bin or lib folder so the dll's are copied to the root will all the other assemblies.  My module translates.  Packaging everything together seems to work.

**Reply by hishamco** [reply]
So is there any problem you facing or it's done?

**Reply by sbwalker** [reply]
@hishamco Oqtane does not currently support the installation of satellite assemblies yet... InstallationManager - InstallPackages() needed to be enhanced to handle subfolders of /bin. I have already made the necessary changes locally and will submit the PR once the other issues in Dev are resolved.

**Reply by hishamco** [reply]
I read your comments before @sbwalker, I will try to did some work in this area, perhaps it's a good feature for Oqtane localization in 2.1.0

**Comment by sbwalker** [comment]
My apologies... I did not mean to confuse you. When you create a nuget package for a custom module, the DLLs are already placed in a /lib folder within the Nuget package. And then when you install the module into Oqtane it copies any files in the /lib folder to the /bin directory of the target system. So all I was asking is if it makes sense for the satellite assemblies to be included in the Nuget package in the same /lib folder structure.

**Reply by leigh-pointer** [reply]
Yes it that makes sense.

Just built a package with the Resource in it checked contents everything fine.

![image](https://user-images.githubusercontent.com/9447612/115285644-66412b00-a14e-11eb-8ff8-d985edad25a7.png)


![image](https://user-images.githubusercontent.com/9447612/115285324-02b6fd80-a14e-11eb-8886-bf31263f2b44.png)

But when I install the package to the server all the files are installed at the root and not into a lib or bin folder.

The module runs fine but no translations.

**Reply by hishamco** [reply]
Did you try to enable auto creating NuGet package for your module from within the VS? I expect the satellite assemblies will be embedded too

**Comment by sbwalker** [comment]
As I mentioned above... Oqtane does not support the installation of satellite assemblies yet... InstallationManager - InstallPackages() needs to be enhanced. My question about how satellite assemblies should be structured in the Nuget package was just to confirm my assumption.
